build-tools/packages/build-cli/src/commands/release/prepare.ts

```diff
@@ -0,0 +1,239 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { Context, Package } from "@fluidframework/build-tools";
+import async from "async";
+import chalk from "chalk";
+import execa from "execa";
+import clonedeep from "lodash.clonedeep";
+import { ResetMode } from "simple-git";
+
+import { packageOrReleaseGroupArg as baseArg, findPackageOrReleaseGroup } from "../../args";
+import { BaseCommand } from "../../base";
+import { Repository, getPreReleaseDependencies } from "../../lib";
+import { isReleaseGroup, ReleaseGroup, ReleasePackage } from "../../releaseGroups";
+
+type CheckFunction = (
+	context: Context,
+	rgOrPkg: ReleaseGroup | ReleasePackage,
+	print?: boolean,
+) => Promise<CheckResult>;
+
+interface CheckResult {
+	result: boolean;
+	message?: string;
+	fixCommand?: string;
+	fatal?: boolean;
+}
+
+const checkNoLocalChanges: CheckFunction = async (context: Context): Promise<CheckResult> => {
+	const status = await context.gitRepo.getStatus();
+	if (status !== "") {
+		return {
+			result: false,
+			message: "There are some local changes. The branch should be clean.",
+			fixCommand: "git stash",
+			fatal: true,
+		};
+	}
+
+	return { result: true };
+};
+
+const checkDepsInstalled: CheckFunction = async (
+	context: Context,
+	rgOrPkg: ReleaseGroup | ReleasePackage,
+	print?: boolean,
+): Promise<CheckResult> => {
+	const packagesToCheck = isReleaseGroup(rgOrPkg)
+		? context.packagesInReleaseGroup(rgOrPkg)
+		: // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		  [context.fullPackageMap.get(rgOrPkg)!];
+
+	async function checkDeps(pkg: Package): Promise<boolean> {
+		const depsAreInstalled = await pkg.checkInstall(print);
+		// console.warn(`deps for ${pkg.name}: ${depsAreInstalled}`);
```

Comment:
We prabably don't want commented out code. If this is something people may want to enable, we should do it in a different way.

Reply:
Good catch; I'll clean this up.